### DISCUSSION TITLE
.gpg-id issues

### DISCUSSION THREAD

**Discussion by dactylroot** [zeapo/Android-Password-Store#2552] [Q&A] [answered]
I'm trying APS-freeRelease-2.0.0-SNAPSHOT.apk on a Pixel 7.

When I try to decrypt a password, I get the message "Found .gpg id, but it contains an invalid key ID, fingerprint or user ID".
I never entered or was prompted for a key ID or secret phrase for my GPG key.

Am I missing an option or setup step?

I also don't see how to remove a GPG key after it has been added.

**Comment by msfjarvis** [comment, marked as answer]
The key ID is read off the repository's `.gpg-id` file same as the `pass` CLI. You can edit the file from your computer and ensure it's not using a short key ID; i.e.: [`3355E9B9`](https://pgp.mit.edu/pks/lookup?op=vindex&search=0xB7843F823355E9B9) is invalid but [`0xB7843F823355E9B9`](https://pgp.mit.edu/pks/lookup?op=vindex&search=0xB7843F823355E9B9) is accepted. You can also use the `FirstName LastName <email@domain.tld>` notation if you prefer that instead.

Re: deleting keys, there is a PGP key management UI that can be found at Settings > PGP settings > Key manager, the keys will be listed here and will have a delete button as well.

**Reply by dactylroot** [reply]
My key has the key ID as required. I am using the same keys I used in previous versions of the Android app. I was not asked for my key passphrase anywhere. I don't currently have the passphrase entry prompt or the delete button under Settings/PGP settings/Key manager. Sorry if that is not in this version, just want to let you know.

**Reply by msfjarvis** [reply]
> My key has the key ID as required.

I'm not sure if we're talking about the same thing?

> I am using the same keys I used in previous versions of the Android app. I was not asked for my key passphrase anywhere.

You were not asked for your passphrase because the app could not parse a valid identifier for what key it should use. If the app doesn't know what key is required, it can't ask for its passphrase.

> I don't currently have the passphrase entry prompt or the delete button under Settings/PGP settings/Key manager. Sorry if that is not in this version, just want to let you know.

If you imported your GPG key your key manager screen should look like this:

<details>
<summary>Screenshot</summary>

![screenshot-20230613-000101](https://github.com/android-password-store/Android-Password-Store/assets/13348378/6f09ac08-5ae2-406d-88d3-d4affcceee78)

</details>

Can I see a screenshot of what you see on your end?

**Reply by dactylroot** [reply]
I now notice a "Delete" button, it's just so tiny that I didn't realize before.

![Screenshot_20230612-113604](https://github.com/android-password-store/Android-Password-Store/assets/1132110/169a68f1-fd94-4cee-8c3d-7841822dfa7f)

I used OpenKeychain with the previous version to associate my keys to the encrypted files. Does that still work now?

**Reply by msfjarvis** [reply]
Oh wow that is a very odd bug, I will make sure it's addressed soon.

APS does not use OpenKeychain anymore since it is no longer maintained.

**Reply by dactylroot** [reply]
my CLI is happy and works fine, but I'm not sure what may be my issue with my .gpg-id file. I re-ran `pass init gpg-id ...` on CLI so I expect it to use the proper format. This produces two lines, first being "gpg-id" and second line being my gpg-id itself.
I tried using the long key, full fingerprint, and then also using my `First Last <email>` format, and they all return this message "Found .gpg id, but it contains an invalid key ID, fingerprint or user ID".

**Reply by msfjarvis** [reply]
The first line being `gpg-id` is what's tripping it up. I've never seen this pattern anywhere but if the pass CLI is fine with it I am not opposed to making the app follow suit. I've made the change in commit e7982f1f0c58, you'll be able to test that in a few minutes with the latest snapshot.

**Reply by dactylroot** [reply]
That snapshot seems good and it identifies the key successfully from the .gpg-id file now.

Unfortunately, I now get "wrong password" when attempting to decrypt. I can confirm the password on desktop CLI with `gpg --export-secret-keys -a <KEYID>` gives me they same key after taking the password.

I have "Encrypt in ASCII armor mode" and "Enable passphrase caching" disabled, but I also tried with each enabled.

Is there something else I can try?

**Reply by msfjarvis** [reply]
Incorrect password shouldn't be tripped incorrectly but I'll have to check the code path again when it's not 2AM to verify that. I'll ping here once I've done my investigation.

**Reply by dactylroot** [reply]
I think the problem was again on my end. Unrelated misuse of `pass` on my CLI and snapshot build works great on my phone now!

Thank you for all your help.
